test/restart/mzcompose.py

```diff
@@ -29,6 +32,44 @@
 ]
 
 
+def workflow_retain_history(c: Composition) -> None:
```

Comment:
Putting this test in `restart` doesn't feel quite right. Is there a better place to put it?

Reply:
`restart` seems right to me, actually! In the future #team-testing will get you a very fast answer to these sorts of questions. They'd probably recommend adding a platform check for this, since platform checks get interspersed with all sorts of disruptions, like restarts. Can do that as a followup though. cc @MaterializeInc/testing for thoughts.